server/server_test.go

```diff
@@ -3324,22 +3324,103 @@ func TestV1Pretty(t *testing.T) {
 func TestPoliciesPutV1(t *testing.T) {
 	t.Parallel()
 
-	f := newFixture(t)
-	req := newReqV1(http.MethodPut, "/policies/1", testMod)
+	v0Module := `package a.b.c
 
-	f.server.Handler.ServeHTTP(f.recorder, req)
+import data.x.y as z
+import data.p
 
-	if f.recorder.Code != 200 {
-		t.Fatalf("Expected success but got %v", f.recorder)
-	}
+q[x] { p[x]; not r[x] }
+r[x] { z[x] = 4 }`
 
-	var response map[string]interface{}
-	if err := json.NewDecoder(f.recorder.Body).Decode(&response); err != nil {
-		t.Fatalf("Unexpected error while unmarshalling response: %v", err)
+	v1Module := `package a.b.c
+
+import data.x.y as z
+import data.p
+
+q contains x if { p[x]; not r[x] }
+r contains x if { z[x] = 4 }`
+
+	tests := []struct {
+		note        string
+		regoVersion ast.RegoVersion
+		module      string
+		expErrs     []string
+	}{
+		{
+			note:        "v0 server, v0 module",
+			regoVersion: ast.RegoV0,
+			module:      v0Module,
+		},
+		{
+			note:        "v0 server, v1 module",
+			regoVersion: ast.RegoV0,
+			module:      v1Module,
+			expErrs:     []string{"var cannot be used for rule name"},
+		},
+		{
+			note:        "v1 server, v1 module",
+			regoVersion: ast.RegoV1,
+			module:      v1Module,
+		},
+		{
+			note:        "v1 server, v0 module",
+			regoVersion: ast.RegoV1,
+			module:      v0Module,
+			expErrs: []string{
+				"`if` keyword is required before rule body",
+				"`contains` keyword is required for partial set rules",
+			},
+		},
```

Comment:
Do we also need test cases for the `--v0-compatible` and `--v1-compatible` paths?

Reply:
Added 👍